tests/test_ui.py

```
@@ -614,47 +519,71 @@ def test_render_table_view_nan_std():
+        m_back.button.return_value = False
+        m_reset.button.return_value = True
+        mock_st_in_steps.columns.return_value = [m_back, m_reset]
 
+        steps._render_footer_actions(["S1"])
+        mock_st_in_steps.rerun.assert_called()
+        assert "interactive_df" not in session_state
+        assert "warm_start_cache" in session_state
 
-def test_steps_render_2_solver_failure_surface():
+
+def test_steps_render_2_solver_failure_surface(mock_streamlit_columns):
```

Ruff (0.15.12)

[warning] Unused function argument: `mock_streamlit_columns`

(ARG001)